src/reconcile/rec_write.c

```diff
@@ -1831,18 +1827,11 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         if (r->page->type != WT_PAGE_ROW_LEAF && chunk->entries == 0)
             return (__wt_set_return(session, EBUSY));
 
-        r->cache_write_hs = true;
-
-        if (r->leave_dirty)
-            r->cache_write_restore = true;
-
-        /* If there are uncommitted changes and no entries were used, copy the disk image. */
-        if (r->leave_dirty && chunk->entries == 0)
-            goto copy_image;
         /* If no entries were used, we have nothing to do. */
-        else if (chunk->entries == 0)
+        if (chunk->entries == 0)
             return (0);
-    }
+    } else if (F_ISSET(r, WT_REC_CHECKPOINT) && chunk->entries == 0)
```

Comment:
What's the case where the `WT_REC_CHECKPOINT` flag isn't set at this point? It seems like checkpoint & eviction share the test `if (chunk->entries == 0)`?